### PR TITLE
Add support for custom tunes

### PR DESCRIPTION
**Pull request by Duncank**
I wanted to be able to execute some custom functions from within the image tool, without creating a completely separate version of this plugin, so I've added support for custom actions (tunes) in the configuration.

### CODE REVIEW

**Comment by neSpecc** [comment]
Nice, we will test it.

**Comment by DavidMunozMartinez** [comment]
I need exactly this, i also was about to re-write the image tool because i want a few more actions in the tool settings. Hope this passes

Thank you!

**Comment by waynehaffenden** [comment]
Any ideas why this hasn't been accepted yet? Is this project even still alive?

**Comment by robonetphy** [comment]
Hi @Duncank!
Thank you for your pull request. Before merging please complete the following things in PR:
- Resolve Conflicts with your PR
- Update version package.json
- Build the image tool using yarn build for production mode

**Comment by Duncank** [comment]
@robonetphy I've updated my changes to the latest version of the plugin and completed your list.

File: src/tunes.js [hunk 1]
```diff
@@ -12,8 +12,9 @@ export default class Tunes {
    * @param {object} tune.api - Editor API
    * @param {Function} tune.onChange - tune toggling callback
    */
-  constructor({ api, onChange }) {
+  constructor({ api, actions, onChange }) {
```
Comment (neSpecc) [inline src/tunes.js:15]: please describe the new parameter in jsdoc

Comment (Duncank) [reply src/tunes.js:16]: I've updated the code!

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/image",
-  "version": "2.4.2",
+  "version": "2.4.3",
```
Comment (robonetphy) [inline package.json:3]: @Duncank Please Change Version 2.4.2 to 2.4.3

**Comment by robonetphy** [comment]
Hi @Duncank,
We have merged your PR but I want to understand the use cases of your custom tunes feature. Since it can't access any internal feature of the image tool how can it be useful to the user using action tool?

**Comment by Duncank** [comment]
Thanks for reaching out; In my application, when a user uploads an image he is presented with an image editor to crop and edit the image. In the custom tunes I also display an 'edit' button to re-open the editor at a later time.
I've also included some toggles to let the user decide whether the caption shows on the webpage or not and stuff like that; for that I use the toggle-functions, coupled with the added classes on the image-block if any extra styling is involved. I want to make some 'position-on-the-left' or '-right' toggles for this as well, using the custom tunes.

**Pull request merged**